threaddb/buckets.go

```diff
@@ -105,6 +300,138 @@ func init() {
 		Name:    buckets.CollectionName,
 		Schema:  bucketsSchema,
 		Indexes: bucketsIndexes,
+		WriteValidator: `
+			if (instance && !instance.version) {
```

Comment:
Here is the spaghetti code `WriteValidator` for a bucket. The logic is not terribly complex, but the code ends up being pretty long winded. 

Note the incline comments!

Reply:
🤯